osbenchmark/workload/params.py

```diff
@@ -797,6 +799,200 @@ def params(self):
         parsed_params.update(self._client_params())
         return parsed_params
 
+
+class VectorSearchParamSource(SearchParamSource):
+    def __init__(self, workload, params, **kwargs):
+        super().__init__(workload, params, **kwargs)
+        self.delegate_param_source = VectorSearchPartitionParamSource(params, self.query_params)
+
+    def partition(self, partition_index, total_partitions):
+        return self.delegate_param_source.partition(partition_index, total_partitions)
+
+    def params(self):
+        raise exceptions.BenchmarkError("Do not use a VectorSearchParamSource without partitioning")
```

Comment:
Might be more appropriate to use `exceptions.WorkloadConfigError` as apposed to catch-all exception `exceptions.BenchmarkError`

Reply:
Updated.